src/app/groups/containers/join-group-confirmation-dialog/join-group-confirmation-dialog.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { DialogModule } from 'primeng/dialog';
+import { ButtonModule } from 'primeng/button';
+import { FormBuilder, FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { SwitchFieldComponent } from '../../../ui-components/collapsible-section/switch-field/switch-field.component';
+import { DatePipe, I18nSelectPipe } from '@angular/common';
+import { GroupApprovals } from '../../data-access/get-group-by-id.service';
+
+export type JoinGroupConfirmationDialogParams = GroupApprovals & { id?: string, name: string };
+export interface JoinGroupConfirmEvent {
+  id?: string,
```

Comment:
It's optional as in future, for join by code, we need only name. Or mb better to move out this param and store it in component?